test/Microsoft.VisualStudio.Composition.Analyzers.Tests/VSMEF012DisallowMefAttributeVersionAnalyzerTests.cs

```diff
@@ -0,0 +1,627 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.VisualStudio.Composition.Analyzers;
+using VerifyCS = CSharpCodeFixVerifier<Microsoft.VisualStudio.Composition.Analyzers.VSMEF012DisallowMefAttributeVersionAnalyzer, Microsoft.VisualStudio.Composition.Analyzers.VSMEF012MigrateToMefV2CodeFixProvider>;
+
+public class VSMEF012DisallowMefAttributeVersionAnalyzerTests
+{
+    private const string EditorConfigDisallowV1 = """
+
+        root = true
+
+        [*.cs]
+        dotnet_diagnostic.VSMEF012.severity = warning
+        dotnet_diagnostic.VSMEF012.allowed_mef_version = V2
+
+        """;
+
+    private const string EditorConfigDisallowV2 = """
+
+        root = true
+
+        [*.cs]
+        dotnet_diagnostic.VSMEF012.severity = warning
+        dotnet_diagnostic.VSMEF012.allowed_mef_version = V1
+
+        """;
```

Comment:
What happens if an unexpected value is assigned to this setting? (e.g. "garbage")
Will the analyzer throw an unhandled exception or report a diagnostic?